### DISCUSSION TITLE
Ctrl + Click hyperlink to automatically open in browser

### DISCUSSION THREAD

**Discussion by Konomi-io** [ppy/osu#30183] [Ideas]
When clicking a link in changelog, it gives you a popup that prompts you to open in browser, or copy the link. ctrl clicking to immediately open a link is common in other programs, and would be nice here. 
though, some other menus open external links without confirmation (notably news and wiki), so i'm not sure if the popup should be there in the first place. nonetheless, if there are places where it happens other than changelog, it would benefit those too

**Comment by peppy** [comment]
Which apps does ctrl-click work in? I've never heard of this one.

**Reply by Konomi-io** [reply]
The biggest one is the Microsoft Office suite. They've basically ingrained "Ctrl+Click to follow link" in my mind, as clicking normally just brings up a bubble with info/actions that apply to the link. This also applies to Google's alternatives like Docs, Sheets, etc. and is what made me think of this suggestion. I'm pretty sure most email clients do something similar to prevent accidental link clicks. 

Web browsers also allow you to Ctrl+Click to open a link in a new tab (same as middle-clicking). While this is a bit different, it's still relevant since it's used to skip an extra click (right click > open in new tab).

I know there are other apps that follow a similar trend, apparently many IDEs and PDF viewers do too. if the prompt appears a lot in lazer, then imo this is widespread enough that implementing it would be worthwhile. Otherwise, the prompt could removed entirely, as it doesn't seem to be consistently applied anyway (which makes it not really that useful).

**Reply by Gabixel** [reply]
Yeah, I think a lot of suite software has that ctrl-click feature (more so because they are WYSIWYGs, so it's common to click on links to select and just edit them). Another example is probably Word and Figma (just to name a few), so even web apps.

I can see it being useful in lazer, for example, if a user wants to quickly open a beatmap in their browser from the chat (for... some reason? maybe to download the map file) instead of opening it in the in-game overlay (which closes the chat(?), and **then** clicking the small "open externally" icon). Probably the same with other in-game links (and/or skip the external link warning, or maybe force it) as the OP mentioned.
Maybe even with user cards (but I think this is risky and maybe too easy to do wrong).

**Reply by peppy** [reply]
The cases you're talking about is ctrl+click to open a url, where normal clicking does nothing, no? And what you're asking for here is ctrl+click to open in browser rather than open in client? They are pretty different flows IMHO.

**Reply by Gabixel** [reply]
Definitely two different cases. Not sure if mine was the best comparison, but I can see this as a kind of shortcut for the scenarios described above. Probably very subtle and rarely used, but something that might feel common/natural to people who use suite software (though, again, it wouldn't make much sense to compare it with lazer).

But yes... in the end, it may actually not make sense as a feature. I don't think it's even implemented in other games, and makes sense to have only one flow here since we're not in some kind of editor.

**Reply by peppy** [reply]
I'd be fine with right click -> open on browser, FWIW.
